DiscordBot/Services/UserService.cs

```diff
@@ -612,6 +613,37 @@ private async Task UserLeft(SocketGuildUser user)
             await _databaseService.DeleteUser(user.Id);
         }
 
+        private async Task AutoCreateThread(SocketMessage messageParam)
+        {
+            if (messageParam.Author.IsBot) return;
+
+            foreach (var prefix in _settings.AutoThreadExclusionPrefixes)
+                if (messageParam.Content.StartsWith(prefix)) return;
+
+            foreach (var AutoThreadChannel in _settings.AutoThreadChannels)
+            {
+                var channel = messageParam.Channel as SocketTextChannel;
+                if (channel.Id.Equals(AutoThreadChannel.Id))
+                {
+                    try
+                    {
+                        var title = AutoThreadChannel.GenerateTitle(messageParam.Author);
+                        var thread = await channel.CreateThreadAsync(title, Discord.ThreadType.PublicThread, Discord.ThreadArchiveDuration.OneDay, messageParam);
+                        if (AutoThreadChannel.CanArchive)
+                        {
+                            var message = await thread.SendMessageAsync(AutoThreadChannel.GenerateFirstMessage(messageParam.Author));
+                            await message.PinAsync();
+                        }
+                    }
+                    catch (Exception err)
+                    {
+                        Console.WriteLine(err);
```

Comment:
Could use ConsoleLogger to make this more obvious, since failure is unlikely it'll stand out more using
``ConsoleLogger.Log(err.ToString(), Severity.Warning);``

Not really important though